wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/TrajectoryCommand.java

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.wpilibj2.command;
+
+import static edu.wpi.first.wpilibj.util.ErrorMessages.requireNonNullParam;
+
+import edu.wpi.first.math.trajectory.Trajectory;
+import edu.wpi.first.wpilibj.Timer;
+import java.util.function.Consumer;
+
+/**
+ * A command that follows a certain trajectory, passing the target state to an output function every
+ * iteration.
+ *
+ * <p>This command does <b>not</b> handle Ramsete, PID, and Feedforward control internally - they
+ * should be used by the subsystem in the output function.
+ */
+public class TrajectoryCommand extends CommandBase {
+  private final Trajectory m_trajectory;
+  private final Consumer<Trajectory.State> m_output;
+  private final Timer m_timer = new Timer();
+
+  /**
+   * Create a new TrajectoryCommand.
+   *
+   * @param trajectory the trajectory to follow.
+   * @param output a function that consumes the target state for each timestep. This state can be
+   *     fed into a Ramsete controller or similar.
+   * @param requirements the subsystems to require.
```

Comment:
The descriptions for parameters are complete sentences: as such, the first letter should be capitalized.